sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/DefaultChangeEncoder.java

```diff
@@ -15,24 +17,58 @@
 import org.apache.cassandra.utils.ByteBufferUtil;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.nio.ByteBuffer;
+import java.util.EnumSet;
 import java.util.Set;
 import java.util.UUID;
 
 class DefaultChangeEncoder implements ChangeEncoder {
+
     private enum Encoding {
         D1,  // delta (version 1 encoding)
         D2,  // delta (version 2 encoding includes tags)
+        D3,  // delta (version 3 encoding, adds change flags to D2
         C1,  // compaction (version 1 encoding)
         A1,  // audit (version 1 encoding)
         H1,  // historical deltas (version 1 encoding)
     }
 
+    private final Encoding _deltaEncoding;
+
+    public DefaultChangeEncoder() {
+        // To support a rolling upgrade the default constructor uses the legacy D2 encoding for new deltas.
+        // Once all Emo instances in an environment have been upgraded a new version should be released where the
+        // default constructor does not use legacy encoding.
+        this(true);
```

Comment:
I thought about an optional yaml config flag which will have legacy D2 and the default as D3 with no config value for the initial release. This way we can just do a ops-release for the second time, but it may not be necessary as we may also not have to go back to D2 in future if everything works fine with D3.